example/app/src/main/java/example/pilgrimsdk/kits/mparticle/com/pilgrimsdkmparticle/activities/MainActivity.kt

```diff
@@ -0,0 +1,90 @@
+
+import android.Manifest
+import android.app.Activity
+import android.content.BroadcastReceiver
+import android.content.Context
+import android.content.Intent
+import android.content.IntentFilter
+import android.content.pm.PackageManager
+import android.os.Bundle
+import android.support.v4.app.ActivityCompat
+import android.support.v7.app.AppCompatActivity
+import com.foursquare.pilgrimsdk.debugging.PilgrimSdkDebugActivity
+import com.mparticle.MParticle
+import example.pilgrimsdk.kits.mparticle.com.pilgrimsdkmparticle.R
+import example.pilgrimsdk.kits.mparticle.com.pilgrimsdkmparticle.utils.isLocationPermissionGranted
+
+class MainActivity : AppCompatActivity() {
+
+    @Suppress("PrivatePropertyName")
+    private val EXAMPLE_REQUEST_LOCATION = 16
+
+    private val receiver =
+        Receiver(this)
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(R.layout.activity_main)
+
+    }
+
+    override fun onResume() {
+        super.onResume()
+        if (checkLocationPermission(this)) {
+            registerMParticleReceiver()
+        }
+    }
+
+    private fun checkLocationPermission(activity: Activity): Boolean {
+        if (!activity.isLocationPermissionGranted()) {
+            // Request fine location permission
+            val pem = arrayOf(Manifest.permission.ACCESS_FINE_LOCATION)
+            ActivityCompat.requestPermissions(activity, pem, EXAMPLE_REQUEST_LOCATION)
+            return false
+        }
+        return true
+    }
+
+    override fun onRequestPermissionsResult(requestCode: Int, permissions: Array<out String>, grantResults: IntArray) {
+        when (requestCode) {
+            EXAMPLE_REQUEST_LOCATION -> {
+                if (grantResults.isNotEmpty() && grantResults[0] == PackageManager.PERMISSION_GRANTED) {
+                    // Location enabled
+                    registerMParticleReceiver()
+                }
+            }
+        }
+        super.onRequestPermissionsResult(requestCode, permissions, grantResults)
+    }
+
+
+    private fun navigateToDebugActivity() {
+        this.unregisterReceiver(receiver)
+        val intent = Intent(this, PilgrimSdkDebugActivity::class.java)
+        startActivity(intent)
+    }
+
+    private fun registerMParticleReceiver() {
+        //Use the BROADCAST_ACTIVE and BROADCAST_DISABLED actions, concatenated with the provider ID
+        val filter = IntentFilter(MParticle.ServiceProviders.BROADCAST_ACTIVE + MParticle.ServiceProviders.PILGRIM_SDK)
+        filter.addAction(MParticle.ServiceProviders.BROADCAST_DISABLED + MParticle.ServiceProviders.PILGRIM_SDK)
+        this.registerReceiver(receiver, filter)
+    }
+
+    class Receiver(private val act: MainActivity) : BroadcastReceiver() {
+        override fun onReceive(context: Context, intent: Intent) {
+            if (intent.action!!.startsWith(MParticle.ServiceProviders.BROADCAST_ACTIVE)) {
```

Comment:
Is there somewhere in here that we should be calling PilgrimSdk.start()?

Reply:
I updated it to call PilgrimSdk.start(context) here

Reply:
the debug activity is just a screen that shows relevant information. Also as a heads up it also wont have anything to display unless `enablePersistentLogs` is called on the builder. We might consider talking to mparticle about recommendations for how to enable debug things remotely.